benchmark-ph1ex1/3D-model/postprocessing.py

```diff
@@ -0,0 +1,236 @@
+    plt.legend(handles=[iref, fuel, oref1, oref2, crod],
+               loc="upper left", bbox_to_anchor=(1.0, 1.0), fancybox=True)
+    plt.savefig("oecd-fullcore-legend", dpi=300, bbox_inches="tight")
+
+
+def global_param(filename1, filename2):
+    '''
+    This function summarizes the global parameters that the benchmark
+    requires to be reported.
+    The results are printed on the terminal.
+
+    Parameters:
+    -----------
+    filename1: [string]
+        name of .csv file with results for case with CR out
+    filename2: [string]
+        name of .csv file with results for case with CR in
+    '''
+
+    file_out = pd.read_csv(filename1)
+    file_in = pd.read_csv(filename2)
+
+    kout = float(file_out['eigenvalue'].tolist()[-1])
+    print('kout: {0}'.format(kout))
+
+    kin = float(file_in['eigenvalue'].tolist()[-1])
+    print('kin: {0}'.format(kin))
+
+    cr_worth = (kout-kin)/(kout*kin)*1e5
+    print('CR worth: {0} pcm'.format(np.round(cr_worth, 4)))
+
+    pow_bottom = float(file_out['power_bottom'].tolist()[-1])
+    print('Power bottom: {0} W'.format(np.round(pow_bottom, 4)))
+
+    pow_top = float(file_out['power_top'].tolist()[-1])
+    print('Power top {0} W'.format(np.round(pow_top, 4)))
+
+    AO = (pow_top - pow_bottom)/(pow_top + pow_bottom)
+    print('Axial offset: {0}'.format(np.round(AO, 4)))
+
+    mem = float(file_out['memory'].tolist()[-1])
+    print('physical memory usage: {0} MiB'.format(mem))
+
+
+def plot_axial_power_distribution(filename, save):
+    '''
+    This function plots the axial power distribution.
+
+    Parameters:
+    -----------
+    filename: [string]
+        name of the .csv file with results
+    save: [string]
+        name of the figure where to save the plot
+    '''
+
+    file = pd.read_csv(filename)
+
+    power = np.zeros(10)
+    for i in range(10):
+        power[i] = float(file['radially_averaged_axial_power_Z'
+                         + str(i+1)].tolist()[-1])
+
+    z = np.linspace(0, 793, 10, endpoint=False)
+    z += 79.3/2
+
+    plt.figure()
+    plt.plot(z, power)
+    plt.ylim(top=8)
+
+    plt.xticks(fontsize=12)
+    plt.yticks(fontsize=12)
+    plt.ylabel(r'Power density [W/cm$^3$]', fontsize=12)
+    plt.xlabel('Axial distance from bottom of active core [cm]', fontsize=12)
+    plt.grid()
+    plt.savefig(save, dpi=300, bbox_inches="tight")
+
+
+def plot_radial_power_distribution(filename, save):
+    '''
+    This function plots the radial power distribution.
+
+    Parameters:
+    -----------
+    filename: [string]
+        name of the .csv file with results
+    save: [string]
+        name of the figure where to save the plot
+    '''
+
+    file = pd.read_csv(filename)
+
+    power = np.zeros(22)
+    for i in range(22):
+        power[i] = float(file['axially_averaged_radial_power_R'
+                         + str(i+1)].tolist()[-1])
+
+    P = 36  # pitch
+    F = P / np.sqrt(3)
+
+    coord = []
+    # 1 - 6
```

Comment:
This section of for loops is programmed in a copy-paste manner that will be a challenge to test and violates DRY.

Reply:
fixed